db-tests/src/test/java/gov/nasa/jpl/aerie/database/PlanCollaborationTests.java

```diff
@@ -621,28 +624,32 @@ void snapshotInheritsAllLatestAsParents() throws SQLException{
 
         final int finalSnapshotId = createSnapshot(planId);
 
-        //assert that there is now only one entry for this plan in plan_latest_snapshot
-        res = statement.executeQuery(
-            """
-            select snapshot_id from plan_latest_snapshot where plan_id = %d;
-            """.formatted(planId));
-        assertTrue(res.first());
-        assertTrue(res.isLast());
-
-        //assert that the snapshot history is n+1 long
-        res = statement.executeQuery(
-            """
-            select get_snapshot_history(%d);
-            """.formatted(finalSnapshotId));
-        assertTrue(res.last());
-        assertEquals(res.getRow(), numberOfSnapshots+1);
-
-        //assert that res contains, in order: finalSnapshotId, snapshotId[0,1,...,n]
-        res.first();
-        assertEquals(res.getInt(1), finalSnapshotId);
-        for (final int snapshotId : snapshotIds) {
+        {
+          //assert that there is now only one entry for this plan in plan_latest_snapshot
+          final var res = statement.executeQuery(
+              """
+                  select count(1) from plan_latest_snapshot where plan_id = %d;
+                  """.formatted(planId));
           res.next();
-          assertEquals(res.getInt(1), snapshotId);
+          assertEquals(1, res.getInt(1));
+        }
+
+        {
+          //assert that the snapshot history is n+1 long
+          final var res = statement.executeQuery(
+              """
+                  select get_snapshot_history(%d);
+                  """.formatted(finalSnapshotId));
+          final var history = new ArrayList<Integer>();
+          while (res.next()) {
+            history.add(res.getInt(1));
+          }
+
+          final var expected = new ArrayList<>();
+          expected.add(finalSnapshotId);
+          expected.addAll(snapshotIds);
+
+          assertEquals(expected, history);
```

Comment:
If this is a combination of `//assert that the snapshot history is n+1 long` and `//assert that res contains, in order: finalSnapshotId, snapshotId[0,1,...,n]`, then the comment should reflect that.

Reply:
Good catch, I'll update the comment :+1:

Reply:
Comment updated